utils/diff_model_converter.py

```diff
@@ -331,7 +331,10 @@ def replace_call_to_super(class_finder: ClassFinder, updated_node: cst.ClassDef,
     original_methods = {f.name.value if hasattr(f, "name") else f: f for f in original_node.body.body}
     updated_methods = {f.name.value if hasattr(f, "name") else f: f for f in updated_node.body.body}
     end_meth = []
-    for name, func in original_methods.items():
+
+    # Iterate directky from node.body as there can be property/setters with same names which are overwritten when we use a dict
+    for func in original_node.body.body:
+        name = func.name.value if hasattr(func, "name") else func
```

Comment:
so you mean original_methods.items() is overwriding some things?

Otherwise the goal is indeed to overwrite the methods using the func.with_changes!

Reply:
This happens only when original class has two methods with identical naming, in my case property and it setter. If we use dict, only one is retained as dict can't hold two keys with same naming.

Not sure if there're cases when directly iterating might cause errors 🤔

Reply:
Hhhhhha I see! 
Then it means we are not taking into account the decorator. Indeed that is a good catch and was probably affectingh @younesbelkada in his cohere porting